addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/CorsScanRule.java

```diff
@@ -0,0 +1,188 @@
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpResponseHeader;
+
+/**
+ * The CORS scan rule identifies Cross-Origin Resource Sharing (CORS) support and overly lenient or
+ * buggy implementations
+ *
+ * @author CravateRouge
+ */
+public class CorsScanRule extends AbstractAppPlugin {
+    private static final Logger LOG = LogManager.getLogger(CorsScanRule.class);
+    private static final String RANDOM_NAME = RandomStringUtils.random(8, true, true);
+    private static final String ACAC = "Access-Control-Allow-Credentials";
+
+    @Override
+    public void scan() {
+        URI uri = getBaseMsg().getRequestHeader().getURI();
+        String authority = uri.getEscapedAuthority();
+        String scheme = uri.getScheme();
+        String handyScheme = scheme + "://";
+
+        // Order of likelihood and severity
+        String[] payloads = {
+            handyScheme + RANDOM_NAME + ".com",
+            "null",
+            handyScheme + RANDOM_NAME + "." + authority,
+            handyScheme + authority + "." + RANDOM_NAME + ".com",
+            // URL encoded backtick used to bypass weak Regex matching only alphanumeric chars to
+            // validate the domain: https://www.corben.io/tricky-CORS/
+            handyScheme + authority + "%60" + RANDOM_NAME + ".com",
+            null,
+            handyScheme + authority
+        };
+
+        boolean secScheme = false;
+        if (scheme == "https") {
+            secScheme = true;
+            payloads[5] = "http://" + authority;
+        }
+
+        for (String payload : payloads) {
+            HttpMessage msg = getNewMsg();
+            msg.getRequestHeader().setHeader(HttpRequestHeader.ORIGIN, payload);
+            try {
+                sendAndReceive(msg);
+
+                HttpResponseHeader respHead = msg.getResponseHeader();
+                String acaoVal = respHead.getHeader(HttpHeader.ACCESS_CONTROL_ALLOW_ORIGIN);
+
+                // If there is an ACAO header an alert will be triggered
+                if (acaoVal == null) {
+                    continue;
+                }
+
+                int risk = Alert.RISK_INFO;
+                boolean vuln = false;
```

Comment:
This is not actually used.

Reply:
It is used when there is a server side header `Access Control Allow Origin` with a value not identified as vulnerable

Reply:
Could you indicate the line where's actually used? I can only see it being written to.

Reply:
I thought you were talking about the `risk` variable. I removed the `vuln` variable in a new commit.